internal/handler/user_service.go

```diff
@@ -0,0 +1,101 @@
+package handler
+
+import (
+	"context"
+
+	"github.com/gobwas/glob"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/smailic05/TaskInfoblox/internal/pb"
+)
+
+type User struct {
+	Address  string
+	Username string
+	Phone    string
+}
+
+type UserService struct {
+	pb.UnimplementedUserServiceServer
+	UserSlice []User
+}
+
+func New() *UserService {
+	return &UserService{UserSlice: make([]User, 0)}
+}
+
+func (s *UserService) AddUser(ctx context.Context, addUser *pb.AddUserRequest) (*pb.AddUserResponse, error) {
+	user := User{
+		Address:  addUser.Address,
+		Username: addUser.Username,
+		Phone:    addUser.Phone}
+	s.UserSlice = append(s.UserSlice, user)
```

Comment:
Is it safe operation? Will that code work as expected if the app processes few requests simultaneously?

Reply:
Does this handle adding the same username/address/phone multiple times?

Reply:
Fixed both